### PR TITLE
Add a couple of examples to the Client doc page

### PR DESCRIPTION
**Pull request by cb372**
I've tried to take a leaf from the [OkHttp docs](http://square.github.io/okhttp/): few words of explanation, just a few lines of code showing how to get something useful done with the client.

### CODE REVIEW

File: docs/src/main/tut/client.md [hunk 1]
```diff
@@ -112,6 +112,56 @@ val greetingsStringEffect = greetingList.map(_.mkString("\n"))
...
+httpClient.expect[String](request)
+```
+
+### Post a form, decoding the JSON response to a case class
+
+```tut:book
+case class AuthResponse(access_token: String)
+
+// See the JSON page for details on how to define this
+implicit val authResponseEntityDecoder: EntityDecoder[IO, AuthResponse] = null
```
[the remark this answers is not in the document]
Comment (cb372) [reply docs/src/main/tut/client.md]: 😬 I've used this trick with tut before. `???` doesn't work but `null` does. I can rewrite the example to not use JSON if this line offends you too much.

File: docs/src/main/tut/client.md [hunk 2]
```diff
@@ -112,6 +112,56 @@ val greetingsStringEffect = greetingList.map(_.mkString("\n"))
...
 
+## Examples
+
+### Send a GET request, treating the response as a string
+
+You can send a GET by calling the `expect` method on the client, passing either
+a String or a `Uri`:
+
+```tut:book
+httpClient.expect[String]("https://google.com/")
```
[the remark this answers is not in the document]
Comment (cb372) [reply docs/src/main/tut/client.md]: OK, I'll remove the String version from the example. It's probably a bit confusing anyway.

I think some explanation of the different ways to construct a `Uri` (i.e. the `uri(...)` macro vs `Uri.fromString`) would also be handy. The docs conveniently use literal strings everywhere so they can use the macro, but that's a bit disingenuous - in real-world code it's more common to be passed a string (e.g. from configuration) and use `fromString`.

Even if the `Uri` stuff is all going to change in the future, I think there's value in documenting how things work right now.

**Comment by cb372** [comment]
@rossabaker Just to clarify, do you mean merge this into master and then open a new PR to backport it to `0.18.x-release`? Or close this PR and retarget it?

File: docs/src/main/tut/client.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (rossabaker) [reply docs/src/main/tut/client.md]: :+1:

File: docs/src/main/tut/client.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rossabaker) [reply docs/src/main/tut/client.md]: I'm just shocked that it works.  I'm okay with it.

**Comment by cb372** [comment]
Superseded by #1800

**Pull request closed without merging**